pkg/cluster/internal/providers/docker/provision.go

```diff
@@ -272,23 +274,39 @@ func generateMountBindings(mounts ...config.Mount) []string {
 }
 
 // generatePortMappings converts the portMappings list to a list of args for docker
-func generatePortMappings(portMappings ...config.PortMapping) []string {
+func generatePortMappings(clusterIPFamily config.ClusterIPFamily, portMappings ...config.PortMapping) []string {
 	args := make([]string, 0, len(portMappings))
 	for _, pm := range portMappings {
-		var hostPortBinding string
-		if pm.ListenAddress != "" {
-			hostPortBinding = net.JoinHostPort(pm.ListenAddress, fmt.Sprintf("%d", pm.HostPort))
-		} else {
-			hostPortBinding = fmt.Sprintf("%d", pm.HostPort)
+		// do provider internal defaulting
+		// in a future API revision we will handle this at the API level and remove this
+		if pm.ListenAddress == "" {
+			switch clusterIPFamily {
+			case config.IPv4Family:
+				pm.ListenAddress = "0.0.0.0" // this is the docker default anyhow
```

Comment:
@liggitt enlightened me about these unspecified addresses, it turns out the go library behaves the same with "0.0.0.0" and "::"

Reply:
dockerd doesn't though, it uses a proxy process for `::` and iptables for `0.0.0.0`